packages/cli/src/server/watcher.rs

```diff
@@ -0,0 +1,259 @@
+use std::{collections::HashSet, path::PathBuf};
+
+use dioxus_cli_config::CrateConfig;
+use dioxus_hot_reload::HotReloadMsg;
+use dioxus_html::HtmlCtx;
+use dioxus_rsx::hot_reload::FileMap;
+use futures_channel::mpsc::{UnboundedReceiver, UnboundedSender};
+use futures_util::StreamExt;
+use notify::{
+    event::{DataChange, ModifyKind},
+    EventKind, FsEventWatcher,
+};
+
+/// This struct stores the file watcher and the filemap for the project.
+///
+/// This is where we do workspace discovery and recursively listen for changes in Rust files and asset
+/// directories.
+pub struct Watcher {
+    tx: UnboundedSender<notify::Event>,
+    rx: UnboundedReceiver<notify::Event>,
+    last_update_time: i64,
+    watcher: FsEventWatcher,
```

Comment:
This should use RecommendedWatcher not FsEventWatcher so that it can build on different platforms e.g. Windows/Linux.

Reply:
Was seeing error compiling on windows.
```
error[E0432]: unresolved import `notify::FsEventWatcher`                                                                                                                                                          
   --> packages\cli\src\server\watcher.rs:11:16
    |
11  |     EventKind, FsEventWatcher,
    |                ^^^^^^^^^^^^^^ no `FsEventWatcher` in the root

```

Made a PR if helpful.
https://github.com/DioxusLabs/dioxus/pull/2596